### ISSUE TITLE
Visualization can't show `info:` links

### ISSUE THREAD

**Issue by cnngimenez** [Kungsgeten/org-brain#30]
Hi!

I have some links in an org file managed by org-brain, these links points to an info page and Emacs help(like `C-h f` functions pages).

When I try to visualize with `org-brain-visualize` it doesn't work. However, I can see some children nodes when trying to enter through its parent.

The message returned by Emacs when trying to display the diagram and summary: 

```
org-brain-insert-resource-button: Wrong type argument: char-or-string-p, (link (:type "info" :path "emacs#Directory" :format plain :raw-link "info:emacs#Directory" :application nil ...))
```

This is de diagram:

```
PINNED:  main


               / emacs
   informática / lenguajes
	|
	Vemacs

wanderlust  org-mode  

-----------------------------------------------

- 
```
This is some of the file contents:

```org
* Variables Por Archivo o Directorio

- Por directorio :: [[info:emacs#Directory%20Variables][info:emacs#Directory Variables]]
- Por archivo :: [[info:emacs#Specifying%20File%20Variables][info:emacs#Specifying File Variables]]
[...]
- Arriba
  - [[help::add-file-local-variable-prop-line]]
    Sino le indicás cuál es el valor, agrega el actual.
  - [[help::delete-file-local-variable-prop-line]]
[...]
* Brainchildren    :brainchildren:
- [[brain:wanderlust][wanderlust]]
- [[brain:org-mode][org-mode]]
```

**Comment by cnngimenez** [comment]
I think that `org-element-map` or `org-element-parse-buffer` creates duplicates when tries to parse `[[info:emacs#Directory%20Variables][info:emacs#Directory Variables]]` creating two elements: one with `info:emacs#Directory%20Variables` and another with `info:emacs#Directory`. Also, it creates an object as the element's contents.

I tried this in the scratch buffer:

```
(with-current-buffer "emacs.org<brain>"
  (setq contenido (org-element-parse-buffer))
  ) ;; Parse all 

(setq glink '())

(org-element-map contenido 'link
  (lambda (link)
    (add-to-list 'glink link)))
(length glink) ;; returns 13
(nth 12 glink) ;; returns the first 
(nth 11 glink) ;; returns the second
```

The problem is that `org-brain-resources` tries to create a list with resources lists, and when one of the copy is processed creates a resouce list with the third element as a list:

```
(setq gres (org-brain-resources "main")) 
(nth 0 gres)
("Variables Por Archivo o Directorio" "info:emacs#Directory%20Variables" (link (:type "info" :path "emacs#Directory" :format plain :raw-link "info:emacs#Directory" :application nil :search-option nil ...)))
```

**Comment by Kungsgeten** [comment]
Yes it seems to be an error (I guess) in `org-element-parse-buffer` when the link description looks like a link on its own.

**Comment by Kungsgeten** [comment]
I posted a bug report to org-mode regarding this. Apparently it has been fixed, but it seems it hasn't been included in the stable version of org-mode yet. They said it will be included in org-mode 9.1, so my suggestion is we wait for that to be released.

**Comment by Kungsgeten** [comment]
`org-mode` 9.1 is now released, and fixes this issue.
